internal/context/upf.go

```diff
@@ -377,9 +378,11 @@ func (upf *UPF) GetUPFID() string {
 }
 
 func (upf *UPF) pdrID() (uint16, error) {
-	if upf.UPFStatus != AssociatedSetUpSuccess {
-		err := fmt.Errorf("UPF[%s] not Associate with SMF", upf.NodeID.ResolveNodeIdToIp().String())
+	select {
```

Comment:
You could create a util function named `IsAssociated()` to remove all of the duplicated parts in this file:
```go
func (upf *UPF) IsAssociated() bool {
	select {
	case <-upf.AssociationContext.Done():
		return false
	default:
	}
	return true
}
```